### PR TITLE
Add table caption documentation

### LINKED ISSUE #9085: [Table Caption] Update documentation

### PR DESCRIPTION
**Pull request by maxbarnas**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Docs (table): Added caption documentation and demo. Closes #9085.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-table/docs/_snippets/features/table-caption.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+						<p style="text-align:center;">Generally good with people</p>
+					</td>
+				</tr>
+				<tr>
+					<td>
+						<p style="text-align:center;">Has a lot of energy</p>
+					</td>
+				</tr>
+			</tbody>
+		</table>
```
Comment (niegowski) [inline packages/ckeditor5-table/docs/_snippets/features/table-caption.html:26]: Maybe some table with more than one column would make more sense?

File: packages/ckeditor5-table/docs/_snippets/features/table.js [hunk 2]
```diff
@@ -21,16 +21,6 @@ ClassicEditor
...
-
-		window.attachTourBalloon( {
-			target: window.findToolbarItem( editor.ui.view.toolbar,
-				item => item.buttonView && item.buttonView.label && item.buttonView.label === 'Insert table' ),
-			text: 'Click to create a table.',
-			editor,
-			tippyOptions: {
-				placement: 'bottom-start'
-			}
-		} );
```
Comment (niegowski) [inline packages/ckeditor5-table/docs/_snippets/features/table.js:33]: Why are you removing this?

**Review by niegowski — CHANGES_REQUESTED** [review]

File: packages/ckeditor5-table/docs/_snippets/features/table-caption.html (hunk 1 above)
Comment (maxbarnas) [reply packages/ckeditor5-table/docs/_snippets/features/table-caption.html:26]: I spent more time trying to find some general data example than I want to admit ;) Adding another column could look nicer, but it's irrelevant to the feature it presents - a table caption. Saying that, I am more than happy to update it if you have better data example to present it here.

File: packages/ckeditor5-table/docs/_snippets/features/table.js (hunk 2 above)
Comment (maxbarnas) [reply packages/ckeditor5-table/docs/_snippets/features/table.js:33]: A tour balloon in a current form can be attached only to elements that exist. A table toolbar is created when the table is focused. We would have to improve the tour balloon function to attach balloons more dynamically.

File: packages/ckeditor5-table/docs/_snippets/features/table-caption.html (hunk 1 above)
Comment (niegowski) [reply packages/ckeditor5-table/docs/_snippets/features/table-caption.html:26]: Maybe some simpler version of what we have in the table manual test? http://fake.ckeditor.com:8125/ckeditor5-table/tests/manual/table.html

File: packages/ckeditor5-table/docs/_snippets/features/table.js (hunk 2 above)
Comment (niegowski) [reply packages/ckeditor5-table/docs/_snippets/features/table.js:33]: But this is not the caption snippet, it's the generic table snippet.

Comment (maxbarnas) [reply packages/ckeditor5-table/docs/_snippets/features/table.js:33]: My bad then. Reverting.

**Review by niegowski — APPROVED** [review]

**Pull request merged**